spec/models/piece_spec.rb

```diff
@@ -1,11 +1,32 @@
-
-
 require 'rails_helper'
 
 RSpec.describe Piece, type: :model do
   let(:game) { FactoryGirl.create(:game) }
   let(:user) { FactoryGirl.create(:user) }
-  let(:piece) { FactoryGirl.create(:piece, :is_on_board) }
+  let(:piece) { FactoryGirl.create(:piece, :is_on_board_black) }
+
+  describe '#move_to!' do
+    context 'when no piece is at destination coordinates' do
+      it 'moves to the destination coordinates' do
+        expect(piece.move_to!(piece, 3, 5)).to have_attributes(column_coordinate: 3, row_coordinate: 5)
+      end
+    end
+
+    context 'when a piece with the opposite color is at the destination coordinates' do
+      it 'moves to new destination coordinates' do
+        piece_white = FactoryGirl.create(:piece, :is_on_board_white)
+        piece.move_to!(piece, 3, 5)
+        piece_white.reload
+        expect(piece).to have_attributes(column_coordinate: 3, row_coordinate: 5)
+      end
+      it 'the other piece is removed from the board' do
+        piece_white = FactoryGirl.create(:piece, :is_on_board_white)
+        piece.move_to!(piece, 3, 5)
+        piece_white.reload
```

Comment:
piece_white still has it's original coordinates.  Do we maybe need to pass it as a parameter?